substrafl/remote/register/manage_dependencies.py

```diff
@@ -0,0 +1,206 @@
+
+    Returns:
+        str: the filename of the wheel
+    """
+    # sys.executable takes the current Python interpreter instead of the default one on the computer
+    try:
+        ret = subprocess.run(
+            [
+                sys.executable,
+                "-m",
+                "pip",
+                "wheel",
+                str(lib_path) + "/",
+                "--no-deps",
+            ],
+            cwd=str(operation_dir),
+            check=True,
+            capture_output=True,
+            text=True,
+        )
+    except subprocess.CalledProcessError as e:
+        raise InvalidUserModuleError from e
+    wheel_name = re.findall(r"filename=(\S*)", ret.stdout)[0]
+
+    return wheel_name
+
+
+def copy_local_wheels(path: Path, dependencies: Dependency) -> List[str]:
+    """Copy the local modules given by the user, generating the wheel if necessary.
+
+    Args:
+        path (Path): the path where the local wheels will be copied.
+        dependencies (Dependency): Dependency object from which the list of local installable dependencies
+            (wheels or modules) will be extracted.
+
+    Returns: list of wheel paths relative to `path`
+    """
+    path.mkdir(exist_ok=True)
+    with tempfile.TemporaryDirectory(dir="/tmp") as tmp_dir_name:
+        tmp_dir = Path(tmp_dir_name)
+        dependencies_paths = dependencies.copy_dependencies_local_package(dest_dir=tmp_dir)
+        wheel_paths = []
+        for dependency in dependencies_paths:
+            if dependency.__str__().endswith(".whl"):
+                wheel_paths.append(str(dependency))
+                shutil.copy(tmp_dir / dependency, path)
+            else:
+                wheel_name = build_user_dependency_wheel(
+                    Path(dependency),
+                    operation_dir=tmp_dir,
+                )
+                wheel_paths.append(wheel_name)
+                shutil.copy(tmp_dir / wheel_name, path)
+    return wheel_paths
+
+
+def local_lib_wheels(lib_modules: List, *, dest_dir: Path) -> List[str]:
+    """Generate wheels for the private modules from lib_modules list and returns the list of names for each wheel.
+
+     It first creates the wheel for each library. Each of the libraries must be already installed in the correct
+     version locally. Use command: ``pip install -e library-name`` in the directory of each library.
+     Then it copies the wheels to the dest_dir.
+
+    This allows one user to use custom version of the passed modules.
+
+    Args:
+        lib_modules (list): list of modules to be installed.
+        dest_dir (str): relative directory where the wheels are saved
+
+    Returns:
+        List[str]: wheel names for the given modules
+    """
+    wheel_names = []
+    dest_dir.mkdir(exist_ok=True, parents=True)
+    for lib_module in lib_modules:
+        lib_path = Path(lib_module.__file__).parents[1]
+        if not (lib_path / "setup.py").exists():
```

Comment:
I'm not sure where those _private modules_ come from ; but would it not be nice to handle the case where `setup.py` are replaced with a `pyproject.toml`?

Reply:
at least worth a comment indeed, I was thinking the same thing at first but it's actually only "our" packages (substra, substrafl and friends) who go through this function